torch/__init__.py

```diff
@@ -779,19 +900,30 @@ def __or__(self, other) -> "SymBool":
     def __sym_not__(self) -> "SymBool":
         raise TypeError("type stub not overridden")
 
-    def __sym_ite__(self, then_val, else_val):
+
+    # booleans behave like fixed-value ints in Python, type checker unable
+    # to handle separate overloads for booleans and ints as a result
+    @_overload
+    def __sym_ite__(self, then_val: _Union["IntLikeType", "BoolLikeType"], else_val: _Union["IntLikeType", "BoolLikeType"]) -> _Union["SymInt", "SymBool"]:
```

Comment:
Technically, bool-like objects should only return a `SymBool` and similarly for int-like objects, but it's not possible to describe this via the Python type system.